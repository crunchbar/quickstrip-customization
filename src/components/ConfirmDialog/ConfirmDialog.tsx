import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import {
  CONFIRM_DIALOG_NO_BTN_ID,
  CONFIRM_DIALOG_YES_BTN_ID,
} from '../../constants/constants';

export interface ConfirmDialogProps {
  open: boolean;
  onClose: () => void;
  onSubmit: () => void;
  description: string;
  title?: string;
}

const ConfirmDialog: React.FC<ConfirmDialogProps> =  ({
  title = 'Confirm',
  description,
  open,
  onClose,
  onSubmit,
}) => (
  <Dialog
    open={open}
    onClose={onClose}
    aria-labelledby="confirm-dialog-title"
    aria-describedby="confirm-dialog-description"
  >
    <DialogTitle id="confirm-dialog-title">{title}</DialogTitle>
    <DialogContent>
      <DialogContentText id="confirm-dialog-description">
        {description}
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onClose} color="primary" id={CONFIRM_DIALOG_NO_BTN_ID}>
        No
      </Button>
      <Button onClick={onSubmit} color="primary" autoFocus id={CONFIRM_DIALOG_YES_BTN_ID}>
        Yes
      </Button>
    </DialogActions>
  </Dialog>
);

export default ConfirmDialog;
