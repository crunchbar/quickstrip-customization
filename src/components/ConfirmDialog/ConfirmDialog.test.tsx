import * as React from 'react';
import {shallow} from 'enzyme';
import ConfirmDialog, {ConfirmDialogProps} from './ConfirmDialog';

const props: ConfirmDialogProps = {
  open: true,
  onClose: () => {},
  onSubmit: () => {},
  description: 'description',
};

describe('<ConfirmDialog />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ConfirmDialog {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
