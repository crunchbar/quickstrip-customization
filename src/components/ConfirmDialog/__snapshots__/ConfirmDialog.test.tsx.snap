// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmDialog /> renders correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-describedby="confirm-dialog-description"
  aria-labelledby="confirm-dialog-title"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="confirm-dialog-title"
  >
    Confirm
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))
      id="confirm-dialog-description"
    >
      description
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      id="confirm-dialog-no-btn"
      onClick={[Function]}
    >
      No
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      color="primary"
      id="confirm-dialog-yes-btn"
      onClick={[Function]}
    >
      Yes
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
